packages/network-subgraphs/src/streamRegistry.ts

```diff
@@ -6,10 +6,12 @@ import { Stream, StreamPermission } from '../generated/schema'
 
 /**
  * Hash the streamId and the userId, in order to get constant-length permission IDs (ETH-867)
- * This avoids indexing problems if the userId is very long (many kilobytes).
+ * This avoids indexing problems if the userId or streamId is very long (many kilobytes).
+ *
+ * TODO: once streamId is no longer possibly very long, remove the slice(0, 1000)
  **/
 function getPermissionId(streamId: string, userId: Bytes): string {
-    return crypto.keccak256(Bytes.fromUTF8(streamId).concat(userId)).toHexString()
+    return streamId.slice(0, 1000) + "-" + crypto.keccak256(Bytes.fromUTF8(streamId).concat(userId)).toHexString()
```

Comment:
It would make sense to make to solve the problem of very long streamIds in a separate PR so that all related changes are easily tracked, and that use unified invariants in the whole code base. With the current code base the `slice()` call in this line is redundant as there is no related slicing e.g. in the `Stream` entity.

It is also possible that we'll solve the issue of very long streamIds somehow differently (without slicing, e.g. by validating the length in the stream creation), and therefore this change add unnecessary point of discontinuity.

```suggestion
    return streamId + "-" + crypto.keccak256(userId).toHexString()
```

Reply:
I don't see how the reduncancy argument follows, as stated, if taken literally. I understand the argument that there is this theoretical streamId length invariant (which isn't true) and according to that invariant this check is redundant. I've considered this argument.

It's possible there's some different solution, but the issue of overlong streamIds can be solved in a separate ticket. This fix is urgent, I believe, and I wouldn't want to undo ETH-867 without a concrete practical reason.

Reply:
Before ETH-867 the permissionId was:
```ts
event.params.streamId + '-' + event.params.user.toHex()
```

So with my suggested change we would keep the streamId part as it was before ETH-867. The root cause why https://github.com/streamr-dev/network-contracts/pull/986 was made was that our _userIds_ were too long, and we needed shorten the permissionId which contained both streamId and userId. Stream IDs haven't caused any problems yet. So minimal change would be to change just the userId part by hashing that and keep the streamId part as-is.

> It's possible there's some different solution, but the issue of overlong streamIds can be solved in a separate ticket. This fix is urgent, I believe, and I wouldn't want to undo ETH-867 without a concrete practical reason.

Yes, that would be ideal way to go. With my suggested code chane this PR doesn't include any code about overlong streamIds and we'll solve the whole issue in another PR just like you mention here 👍 

> I don't see how the reduncancy argument follows

The redundancy comes from the fact that currently our `Stream` entity can't contain overlong id field anyway. If we'd try to store such `Stream` entity, the operation would fail. As each `StreamPermission` belongs to one `Stream`, this slicing is needed only if this permission belongs to an overlong `Stream`. But as there can't be such streams in the subgraphs at the moment, the reference would in invalid anyway. Therefore the handling of this kind of impossible scenario is redundant.

By unified invariants in the whole code base I meant that it is easier to understand and manage a code base where some facts/restrictions/etc. are applied to the whole code base instead of a subset of the code base. I.e. it is easy to define that we don't currently support overlong stream ids, and all code is implemented with that restriction in mind. Or we could define that we need to support overlong stream ids, and fix all issues related to that functionality. But the middle ground, where each code snippet has different level of support for that feature is most difficult to manage.